File: code/dobot_quad_sdk-main/high_level/python/examples/e5_robot_state.py
#!/usr/bin/env python3
"""Example 5: Get full robot state telemetry."""
import sys
from dobot_quad import RobotClient


def main():
    robot = RobotClient(sys.argv[1] if len(sys.argv) > 1 else "192.168.5.2:50051")
    res = robot.get_state()

    if not res.success:
        print(f"Failed: {res.message}")
        return 1

    print(f"Current state: {res.current_state}\n")
    print(f"Speed ratio: {res.current_speed_ratio}")
    print(f"Obstacle avoidance: {'enabled' if res.obstacle_avoidance_enabled else 'disabled'}\n")

    s = res.robot_state
    fmt = lambda arr: [f"{x:.2f}" for x in arr]

    print("Leg Joints:")
    print(f"  pos [rad]:     {fmt(s.jpos_leg)}")
    print(f"  pos_des [rad]: {fmt(s.jpos_leg_des)}")
    print(f"  vel [rad/s]:   {fmt(s.jvel_leg)}")
    print(f"  vel_des:       {fmt(s.jvel_leg_des)}")
    print(f"  tau [Nm]:      {fmt(s.jtau_leg)}")
    print(f"  tau_des [Nm]:  {fmt(s.jtau_leg_des)}")

    print(f"\nBody:")
    print(f"  pos [m]:       {fmt(s.pos_body)}")
    print(f"  vel [m/s]:     {fmt(s.vel_body)}")
    print(f"  acc [m/s^2]:   {fmt(s.acc_body)}")
    print(f"  omega [rad/s]: {fmt(s.omega_body)}")
    print(f"  rpy [rad]:     {fmt(s.ori_body)}")

    print(f"\nContact:")
    print(f"  left  [N]:     {fmt(s.grf_left)}")
    print(f"  right [N]:     {fmt(s.grf_right)}")
    print(f"  vertical [N]:  {fmt(s.grf_vertical_filtered)}")

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: code/dobot_quad_sdk-main/high_level/cpp/e5_robot_state.cpp
// Example 5: Get full robot state telemetry.
// Usage: ./e5_robot_state [server_address]

#include "robot_client.h"
#include <iomanip>

int main(int argc, char** argv)
{
    robot::Client client(argc > 1 ? argv[1] : "192.168.5.2:50051");
    auto res = client.get_state();

    if (!res.success()) {
        std::cerr << "Failed: " << res.message() << std::endl;
        return 1;
    }

    std::cout << "Current state: " << res.current_state() << "\n" << std::endl;
    std::cout << "Speed ratio: " << res.current_speed_ratio() << std::endl;
    std::cout << "Obstacle avoidance: " << (res.obstacle_avoidance_enabled() ? "enabled" : "disabled") << "\n"
              << std::endl;

    const auto& s = res.robot_state();
    auto print = [](const google::protobuf::RepeatedField<float>& a, const std::string& label) {
        std::cout << label << ": [";
        for (int i = 0; i < a.size(); ++i) {
            if (i)
                std::cout << ", ";
            std::cout << std::fixed << std::setprecision(2) << a[i];
        }
        std::cout << "]" << std::endl;
    };

    std::cout << "Leg Joints:" << std::endl;
    print(s.jpos_leg(), "  pos [rad]");
    print(s.jpos_leg_des(), "  pos_des [rad]");
    print(s.jvel_leg(), "  vel [rad/s]");
    print(s.jvel_leg_des(), "  vel_des [rad/s]");
    print(s.jtau_leg(), "  tau [Nm]");
    print(s.jtau_leg_des(), "  tau_des [Nm]");

    std::cout << "\nBody:" << std::endl;
    print(s.pos_body(), "  pos [m]");
    print(s.vel_body(), "  vel [m/s]");
    print(s.acc_body(), "  acc [m/s^2]");
    print(s.omega_body(), "  omega [rad/s]");
    print(s.ori_body(), "  rpy [rad]");

    std::cout << "\nContact:" << std::endl;
    print(s.grf_left(), "  left [N]");
    print(s.grf_right(), "  right [N]");
    print(s.grf_vertical_filtered(), "  vertical [N]");

    return 0;
}
